### PR TITLE
Enki Cleanup 

### PR DESCRIPTION
**Pull request by sparksam**
This PR is part of the `enki` cleanup task.
- [x] Optimized the validations
- [x] Accept both directories and single files for validation.

### CODE REVIEW

File: src/org/openstax/xml/Main.java [hunk 1]
```diff
@@ -3,27 +3,66 @@
...
+                // For each file, run the checks
+                System.out.println("Validating " + file);
+                runChecks(file, desiredChecks);
+            }
         }
-        String[] desiredChecks = args.length == 1 ? new String[]{"all"} : Arrays.copyOfRange(args, 1, args.length);
-        runChecks(input, desiredChecks);
+    }
+
+    public static Set<String> getResources(String path, String pattern) throws Throwable {
```
Comment (philschatz) [inline src/org/openstax/xml/Main.java:48]: Would something like [Files.find](https://stackoverflow.com/a/38102485) work?

**Review by philschatz — COMMENTED** [review]
Just a small suggestion

File: src/org/openstax/xml/Main.java (hunk 1 above)
Comment (sparksam) [reply src/org/openstax/xml/Main.java:48]: Great suggestion. I will implement that.

**Review by philschatz — APPROVED** [review]
:tada:

**Pull request merged**